taskcluster_util/taskcluster_login.py

```diff
@@ -115,20 +115,23 @@ def do_GET(self):
                 else:
                     query_components[k] = v
             logger.debug('GET Parameters: {}'.format(query_components))
-            # write to file
-            logger.info('Write credentials to {}'.format(credentials_file))
-            with open(credentials_file, mode='w') as f:
-                json.dump(query_components, f, indent=4)
-            # response
-            self.send_response(200)
-            self.send_header('Content-type', 'text/html')
-            self.end_headers()
-            self.wfile.write('<h1>Login Successful</h1><br>You can close this window now...')
-            # stop server
-            logger.debug('======= Stop Server =======')
-            assassin = threading.Thread(target=self.server.shutdown)
-            assassin.daemon = True
-            assassin.start()
+            if 'certificate' not in query_components:
+                logger.warning('Do not get certificate from GET Parameters.'.format(query_components))
+            else:
+                # write to file
+                logger.info('Write credentials to {}'.format(credentials_file))
```

Comment:
I guess we can make `do_GET()` less complex by splitting these 13 lines into 3:
```py
else:
    self.credentials_file_handler.save(content=query_components)
    self.respond_successful_login()
    self.stop()
```

Then you define these functions like:
```py
class ServerHandler(BaseHTTPRequestHandler):
    # ...
    def respond_successful_login(self):
           self.send_response(200)
           self.send_header('Content-type', 'text/html')
           self.end_headers()
           self.wfile.write('<h1>Login Successful</h1><br>You can close this window now...')

    def stop(self):
        logger.debug('======= Stop Server =======')
        assassin = threading.Thread(target=self.server.shutdown)
        assassin.daemon = True
        assassin.start()
```
```py
class CredentialsFileHandler:
    def __init__(self, file_path=None):
        self.file_path = file_path if file_path is not None else os.path.join(os.path.expanduser('~'), 'tc_credentials.json')

    def save(self, content):
        with open(self.file_path, mode='w') as f:
            json.dump(content, f, indent=4)
    
    def remove_exist_file(self):
        # put the content of _remove_exist_file() in here
```

What do you think?